### PR TITLE
sdm update

### PR DESCRIPTION
**Pull request by babaknaimi**
Given the major update of the package, I updated the yml file. In particular, new features of the sdm package including supporting bias incorporation in background generation, collinearity check, etc. are included. Some of the other features (e.g., supporting multi-species, and occ_cleaning, etc.) were wrongly assigned to no, and they are revised now.

### CODE REVIEW

**Comment by sylvainschmitt** [comment]
Thanks @babaknaimi. To help review, can you precise the addition (functions and or new functionality) in terms of:
* occ_cleaning: Function(s) to clean occurrence data (remove errors, fix georeferencing, etc.; yes/no)
* env_collinearity: Function(s) to explore or address collinearity of environmental data (yes/no)
* bias: Function(s) to assess and/or correct biases in data (e.g., spatially thinning occurrences or weighting background to address occurrence sampling bias; yes/no)
* mod_multispecies: Function(s) that model multiple species at once to estimate community composition / diversity (i.e., not a “stacked model; yes/no)
* metadata: Function(s) to record analysis metadata (yes/no)
Thanks in advance

**Comment by jamiemkass** [comment]
@babaknaimi  Hello again -- can you please specify why you turned on these characteristics? Which functions in your package do these things?

**Comment by babaknaimi** [comment]
> @babaknaimi Hello again -- can you please specify why you turned on these characteristics? Which functions in your package do these things?

Hello,

occ_cleaning (yes): the data cleaning is part of the sdmData function to prepare data for sdm (modelling) that checks for issues including missing information (can either impute the missing records or remove), remove duplications, fix georeference mismatch between species and environmental data.

env_collinearity (yes): In the sdmData function, the user can specify in the formula the argument to test collinearity and select non-colinear variables. example: d <- sdmData(sp~ . + select(., method="vifstep")....) 

bias (yes): In the function background (also in sdmData, bg argument), there is an option for introducing a bias layer to select weighted background records incorporating the bias layer.

mod_multispecies (yes): It is possible to fit models for multiple-species at once using the sdm function.

metadata (yes):  In the sdmData function, sdm method definition (add function), it is possible to include metadata information. a slot of sdmdata class (and .method container class) is a specific class (.metadata) designed to keep metadata records.

Hope this information clarifies the changes.

Best wishes,
Babak

**Comment by sylvainschmitt** [comment]
Great thank you @babaknaimi !

@jamiemkass , regarding mod_multispecies, I think the idea was to have functions doing more thatn simple stacking:

> Function(s) that model multiple species at once to estimate community composition / diversity (i.e., not a “stacked model; yes/no)

So should we keep it as is, and what about `sdm` in this case, or do you wish to extend it?

**Comment by babaknaimi** [comment]
> Great thank you @babaknaimi !
> 
> @jamiemkass , regarding mod_multispecies, I think the idea was to have functions doing more thatn simple stacking:
> 
> > Function(s) that model multiple species at once to estimate community composition / diversity (i.e., not a “stacked model; yes/no)
> 
> So should we keep it as is, and what about `sdm` in this case, or do you wish to extend it?

Hi, sorry for the confusion. The current version of the package only supports organising and handling multiple species data in a single sdmdata object and fit/predict the models for multiple species simultaneously that can be used for functions like niche similarity, or having a stack of predicted/ensembled layers for multiple species but still no function to calculate diversity metrics in the package. In the next major update, the plan is to include many community-based metrics (including diversity). So please feel free to change this factor to no.

**Comment by sylvainschmitt** [comment]
> > Great thank you @babaknaimi !
> > @jamiemkass , regarding mod_multispecies, I think the idea was to have functions doing more thatn simple stacking:
> > > Function(s) that model multiple species at once to estimate community composition / diversity (i.e., not a “stacked model; yes/no)
> > 
> > 
> > So should we keep it as is, and what about `sdm` in this case, or do you wish to extend it?
> 
> Hi, sorry for the confusion. The current version of the package only supports organising and handling multiple species data in a single sdmdata object and fit/predict the models for multiple species simultaneously that can be used for functions like niche similarity, or having a stack of predicted/ensembled layers for multiple species but still no function to calculate diversity metrics in the package. In the next major update, the plan is to include many community-based metrics (including diversity). So please feel free to change this factor to no.

Ok thanks for precising it @babaknaimi . However it seems unfair that I would be the judged of that and it could be great if someone was ok to review the changes else than me. @jamiemkass @hannahlowens @adamlilith , would you be available?

Similarly, @babaknaimi , when @ecospat will have precised is submission, you would be available to quickly review it?

Thanks, _Sylvain_

File: inst/extdata/packages/sdm.yaml [hunk 1]
```diff
@@ -1,31 +1,31 @@
...
-mod_multispecies: no
+mod_multispecies: yes
 mod_mechanistic: no
 pred_general: yes
 pred_extrapolation: yes
 pred_inspect: yes
 post_processing: yes
 gui: yes
-metadata: no
+metadata: yes
```
Comment (jamiemkass) [inline inst/extdata/packages/sdm.yaml]: If the pkg is not explicitly making metadata for the user, I don't think we can say "metadata = yes". It seems there is a slot for the user to put their own metadata in, which is great, but if the pkg itself doesn't make it, I'd say "metadata = no".

File: inst/extdata/packages/sdm.yaml [hunk 2]
```diff
@@ -1,31 +1,31 @@
...
+bias: yes
 study_region: yes
 backg_sample: yes
 data_partitioning: yes
 mod_fit: yes
 mod_ensemble: yes
 mod_stack: no
 mod_evaluate: yes
-mod_multispecies: no
+mod_multispecies: yes
```
Comment (jamiemkass) [inline inst/extdata/packages/sdm.yaml]: I'd say "multispecies = no" here based on comments from @babaknaimi.

**Review by jamiemkass — CHANGES_REQUESTED** [review]

File: inst/extdata/packages/sdm.yaml (hunk 1 above)
Comment (sylvainschmitt) [reply inst/extdata/packages/sdm.yaml]: @babaknaimi , could you precise?

**Comment by sylvainschmitt** [comment]
Thanks @jamiemkass .

**Pull request merged**